src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -288,10 +288,18 @@ public void addExternalTrigger(ExternalTrigger external_trigger) {
 	}
 
 	@SuppressWarnings("unused") public String[] getListOf(String type) {
-		return getListOfForWorkspace(mcreator.getWorkspace(), type);
+		return getListOfForWorkspace(mcreator.getWorkspace(), type, null);
+	}
+
+	@SuppressWarnings("unused") public String[] getListOfFrom(String type, String source) {
+		return getListOfForWorkspace(mcreator.getWorkspace(), type, source);
 	}
 
 	@SuppressWarnings("unused") public static String[] getListOfForWorkspace(Workspace workspace, String type) {
+		return getListOfForWorkspace(workspace, type, null);
+	}
+
+	@SuppressWarnings("unused") public static String[] getListOfForWorkspace(Workspace workspace, String type, String source) {

```

Comment:
I don't know what source here is supposed to mean and other users won't either. Source as source code, source as source of type, or? Add some JavaDOC to make it clear for other users.

If this is specific to this entry (which is seems is), I would much prefer having a separate method for this than hijacking some other-purpose method with very specific argument

Reply:
Is it fine if I incorporate the list that needs this parameter into `BlocklyJavascriptBridge.openEntrySelector()` and use `customEntryProviders` to specify the value of the parameter?

Reply:
:heavy_check_mark: Fixed in case AlsoSomeoneElse approves the new approach :eyes:

Reply:
See my comment above, your changes to the system don't seem necessary to me and don't relate to the PR which adds custom entity data parameters, not changes blockly data list system

Reply:
Without these changes I don't see how procedure blocks added here are possible :man_shrugging:

Reply:
Closing this one as my comment relates to the JS code